src/entity.cpp

```diff
@@ -9215,6 +9250,84 @@ void Entity::setRangedProjectileAttack(Entity& marksman, Stat& myStats)
 	}
 }
 
+/* entity.cpp
+ * Attempts to set the Entity on fire. Entities that are not Burnable or are already on fire will return before any processing
+ * Entities that do not have Stats (such as furniture) will return after setting the fire time and chance to stop at max
+ * Entities with Stats will have their fire time (char_fire) and chance to stop being on fire (chanceToPutOutFire) reduced by their CON
+ * Calculations for reductions is outlined in this function
+ */
+void Entity::SetEntityOnFire()
+{
+	// Check if the Entity can be set on fire
+	if ( this->flags[BURNABLE] )
+	{
+		// Check if the Entity is already on fire
+		if ( !(this->flags[BURNING]) )
+		{
+			this->flags[BURNING] = true;
+			serverUpdateEntityFlag(this, BURNING);
+
+			/* Set the time the Entity will be on fire, based off their CON
+			 * |\_ MAX_TICKS_ON_FIRE is reduced by every 2 points in CON
+			 * |
+			 * |\_ Fire has a minimum of 2 seconds (100 ticks), and a maximum of 20 seconds (1000 ticks)
+			 * |  \_ Constants are defined in entity.hpp: MIN_TICKS_ON_FIRE and MAX_TICKS_ON_FIRE
+			 * |
+			 *  \_ For every 5 points of CON, the chance to stop being on fire is reduced
+			 *    \_ The chance to stop being on fire has a minimum of 1 in 10, and a maximum of 1 in 5
+			 *      \_ Constants are defined in entity.hpp: MIN_CHANCE_STOP_FIRE and MAX_CHANCE_STOP_FIRE
+			 */
+
+			// Set the default time on fire
+			this->char_fire = MAX_TICKS_ON_FIRE;
```

Comment:
I'm leaning towards enforcing at least 2-5 damage ticks before having a chance to stop fire.. Could increase the base chance to stop from 1 in 10 to 1 in 8 to compensate or something. It's a weird effect since it could stop instantly but go on forever